maintainers/scripts/haskell/regenerate-hackage-packages.sh

```diff
@@ -0,0 +1,19 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p coreutils haskellPackages.cabal2nix-latest -I nixpkgs=.
+
+# This script is used to regenerate nixpkgs' Haskell package set, using a tool
+# called hackage2nix. hackage2nix looks at
+# pkgs/development/haskell-modules/configuration-hackage2nix.yaml and generates
+# a Nix expression for package version specified there, using the Cabal files
+# from the Hackage database (available under all-cabal-hashes) and its
+# companion tool cabal2nix.
+#
+# Related scripts are update-hackage.sh, for updating the snapshot of the
+# Hackage database used by hackage2nix, and update-cabal2nix-latest.sh,
+# for updating the version of hackage2nix used to perform this task.
+
+set -euo pipefail
+
+unpacked_hackage="$(nix-build -E 'with import ./. {}; runCommand "unpacked-cabal-hashes" { src = all-cabal-hashes; } "unpackPhase ; mv $sourceRoot $out"' --no-out-link)"
```

Comment:
This should unpack directly to $out instead of using the unpackPhase, since it takes almost twice as long otherwise due to having to move all files into the store.

Reply:
Good catch.

Reply:
@hyperfekt Does the change I pushed look better?

Reply:
Yes, that should work well.